skimage/future/manual_segmentation.py

```diff
@@ -0,0 +1,216 @@
+
+            plt.draw()
+
+    fig.canvas.mpl_connect('button_press_event', _extend_polygon)
+
+    plt.show(block=True)
+
+    labels = (_mask_from_vertices(vertices, image.shape[:2], i)
+              for i, vertices in enumerate(list_of_vertex_lists, start=1))
+    if return_all:
+        return np.stack(labels)
+    else:
+        return reduce(np.maximum, labels, np.broadcast_to(0, image.shape[:2]))
+
+
+def manual_lasso_segmentation(image, alpha=0.4, return_all=False):
+    """Return a label image based on freeform selections made with the mouse.
+
+    Parameters
+    ----------
+    image : (M, N[, 3]) array
+        Grayscale or RGB image.
+
+    alpha : float, optional
+        Transparency value for polygons drawn over the image.
+
+    return_all : bool, optional
+        If True, an array containing each separate polygon drawn is returned.
+        (The polygons may overlap.) If False (default), later polygons
+        "overwrite" earlier ones where they overlap.
+
+    Returns
+    -------
+    labels : array of int, shape ([Q, ]M, N)
+        The segmented regions. If mode is `'separate'`, the leading dimension
+        of the array corresponds to the number of regions that the user drew.
+
+    Notes
+    -----
+    Press and hold the left mouse button to draw around each object.
+
+    Examples
+    --------
+    >>> from skimage import data, future, io
+    >>> camera = data.camera()
+    >>> mask = future.manual_lasso_segmentation(camera)  # doctest: +SKIP
+    >>> io.imshow(mask)  # doctest: +SKIP
+    >>> io.show()  # doctest: +SKIP
+    """
+    list_of_vertex_lists = []
+    polygons_drawn = []
+    patch_objects = []
+
+    if image.ndim not in (2, 3):
+        raise ValueError('Only 2D grayscale or RGB images are supported.')
+
+    fig, ax = plt.subplots()
+    ax.imshow(image, cmap="gray")
+    ax.set_axis_off()
+
+    def _undo(*args, **kwargs):
+        if list_of_vertex_lists:
+            list_of_vertex_lists.pop()
+            # Remove last polygon from list of polygons...
+            last_poly = polygons_drawn.pop()
+            # ... then from the plot
+            last_poly.remove()
+
+    undo_pos = plt.axes([0.85, 0.05, 0.075, 0.075])
+    undo_button = matplotlib.widgets.Button(undo_pos, u'\u27F2')
+    undo_button.on_clicked(_undo)
+
+    def _on_lasso_selection(vertices):
+        if len(vertices) < 3:
+            return
+        list_of_vertex_lists.append(vertices)
+        polygon_object = _draw_polygon(ax, vertices, alpha=alpha)
+        polygons_drawn.append(polygon_object)
+        plt.draw()
+
+    lasso = matplotlib.widgets.LassoSelector(ax, _on_lasso_selection)
+
+    plt.show(block=True)
+
+    labels = (_mask_from_vertices(vertices, image.shape[:2], i)
+              for i, vertices in enumerate(list_of_vertex_lists, start=1))
+    if return_all:
+        return np.stack(labels)
+    else:
+        return reduce(np.maximum, labels, np.broadcast_to(0, image.shape[:2]))
```

Comment:
@soupault the point is that the multi-labels are useful, not that the overlap is useful. And they are *very* useful: if you want to segment 10 objects in an image, for example. It's much more annoying adding the extra step of collapsing the masks, or labelling them (in which case you have to be really really careful that they don't touch), than it is to add a >0 if you need it. Which you probably don't because functions that deal with segmentations deal with integer labels. (And I think it's worth having that be consistent throughout the library.)